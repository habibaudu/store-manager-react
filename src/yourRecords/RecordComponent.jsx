import React from 'react';
import { Link } from 'react-router-dom';

class RecordComponent extends React.Component {
  componentDidMount() {
    this.props.fetchMyRecords();
  }

  render() {
    return (
      <div>
        <nav>
          <div className="logo">
            <h4>Store Manager</h4>
          </div>
          <ul className="nav-links">
            <li>
              {' '}
              <Link to="/">Home</Link>
            </li>
            <li>
              {' '}
              <Link to="/products">products</Link>
            </li>
            <li>
              {' '}
              <Link to="/records">Records</Link>
            </li>
          </ul>
          <div className="burger">
            <div className="line1" />
            <div className="line2" />
            <div className="line3" />
          </div>
        </nav>

        <div className="addbox">
          <div className="log_head">Your Records</div>
          <div style={{ overflowX: 'auto' }}>
            {this.props.record.records ? (
              <table id="myrecords">
                <thead>
                  <tr>
                    <th>salesID</th>
                    <th>Sold By</th>
                    <th>Product Name</th>
                    <th>Price Each</th>
                    <th>Quantity Sold</th>
                    <th>Price Sold</th>
                    <th>Total Amount</th>
                    <th>Date Sold</th>
                  </tr>
                </thead>
                <tbody>
                  {this.props.record.records.rows.map(record => {
                    const {
                      id,
                      productname,
                      price,
                      attendants_id,
                      totalprice,
                      total,
                      quantity,
                      created_date,
                    } = record;
                    return (
                      <tr key={productname}>
                        <td>{id}</td>
                        <td>{attendants_id}</td>
                        <td>{productname}</td>
                        <td>{price}</td>
                        <td>{quantity}</td>
                        <td>${total}</td>
                        <td>${totalprice}</td>
                        <td>{created_date}</td>
                      </tr>
                    );
                  })}
                </tbody>
              </table>
            ) : (
              <h3>loading records please wait...</h3>
            )}

            <div className="form_footer">
              <p>Store Manager copyright &copy 2018 </p>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default RecordComponent;
