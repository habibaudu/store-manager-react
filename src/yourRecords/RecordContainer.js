import { connect } from 'react-redux';
import { fetchMyRecords} from './duck';
import RecordComponent  from './RecordComponent';


const mapStateToProps = (state) => {
  return {
          record :state.recordReducer,

        }
};

const mapDispatchToProps = dispatch => {
  return {
    fetchMyRecords: () =>
      dispatch(fetchMyRecords()),
  };
};
const RecordContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(RecordComponent);

export { RecordContainer, mapDispatchToProps, mapStateToProps };
