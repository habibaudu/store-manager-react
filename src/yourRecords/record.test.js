import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import axios from 'axios';
import RecordComponent from './RecordComponent';
import { RecordContainer, mapDispatchToProps } from './RecordContainer';
import {
  actions,
  types,
  constants,
  recordReducer,
  fetchMyRecords,
} from './duck';

jest.mock('axios');

const { setAllRecords } = actions;
const props = {
  fetchMyRecords: jest.fn(),
  record: {
    records: {
      rows: [
        {
          id: 2,
          attendants_id: 3,
          productname: 'shoe',
          price: 2345,
          quantity: 34,
        },
        {
          id: 22,
          attendants_id: 4,
          productname: 'shoe',
          price: 2000,
          quantity: 34,
        },
      ],
    },
  },
};

describe('SALES ATTENDANT RECORD TEST SUITE', () => {
  describe('Record Component', () => {
    it('should render the record Page', () => {
      const component = shallow(<RecordComponent {...props} />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });

    it('should render the Record container', () => {
      const component = shallow(<RecordContainer />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });
  });

  it('should ensures mapDispatchToProps dispatches the specified actions', () => {
    const mock = configureStore([thunk]);
    const store = mock(props);
    const component = shallow(
      <Provider store={store}>
        <RecordContainer />
      </Provider>,
    );
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).fetchMyRecords();
  });
});

describe('TEST FOR FETCH SALES ATTENDANT RECORD', () => {
  const payload = {
    data: {
      productname: 'productname',
      id: 1,
      quantity: 1,
      minimum: 2,
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore({});
  let wrapper;
  const response = { message: 'fetchall records' };
  axios.get.mockResolvedValue(response);
  store.dispatch(fetchMyRecords()).then(() => {
    store.dispatch(setAllRecords(payload));
  });
});

describe('Record Reducers', () => {
  it('should setup default state values', () => {
    const state = recordReducer(undefined, {
      type: '@@INIT',
    });
    expect(state).toEqual({});
  });

  it('it should change the load record  state', () => {
    const action = {
      type: types.LOAD_ALL_RECORD,
      records: '5 records',
    };
    const state = recordReducer(undefined, action);
    expect(state.records).toEqual(action.records);
  });
});
