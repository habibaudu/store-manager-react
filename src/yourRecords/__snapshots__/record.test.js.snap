// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SALES ATTENDANT RECORD TEST SUITE Record Component should render the Record container 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SALES ATTENDANT RECORD TEST SUITE Record Component should render the record Page 1`] = `
<div>
  <nav>
    <div
      className="logo"
    >
      <h4>
        Store Manager
      </h4>
    </div>
    <ul
      className="nav-links"
    >
      <li>
         
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
         
        <Link
          replace={false}
          to="/products"
        >
          products
        </Link>
      </li>
      <li>
         
        <Link
          replace={false}
          to="/records"
        >
          Records
        </Link>
      </li>
    </ul>
    <div
      className="burger"
    >
      <div
        className="line1"
      />
      <div
        className="line2"
      />
      <div
        className="line3"
      />
    </div>
  </nav>
  <div
    className="addbox"
  >
    <div
      className="log_head"
    >
      Your Records
    </div>
    <div>
      <table
        id="myrecords"
      >
        <thead>
          <tr>
            <th>
              salesID
            </th>
            <th>
              Sold By
            </th>
            <th>
              Product Name
            </th>
            <th>
              Price Each
            </th>
            <th>
              Quantity Sold
            </th>
            <th>
              Price Sold
            </th>
            <th>
              Total Amount
            </th>
            <th>
              Date Sold
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="shoe"
          >
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              shoe
            </td>
            <td>
              2345
            </td>
            <td>
              34
            </td>
            <td>
              $
            </td>
            <td>
              $
            </td>
            <td />
          </tr>
          <tr
            key="shoe"
          >
            <td>
              22
            </td>
            <td>
              4
            </td>
            <td>
              shoe
            </td>
            <td>
              2000
            </td>
            <td>
              34
            </td>
            <td>
              $
            </td>
            <td>
              $
            </td>
            <td />
          </tr>
        </tbody>
      </table>
      <div
        className="form_footer"
      >
        <p>
          Store Manager copyright &copy 2018 
        </p>
      </div>
    </div>
  </div>
</div>
`;
